### PR TITLE
Allow images to be bundled within the walkthroughs themselves

### PR DESCRIPTION
**Pull request by aidenkeating**
Currently images are included in a separate images directory which
is provided by the server. However, when users want to add their
own walkthroughs they won't be able to add images directly to this
repo. It'd also mean that their adoc wouldn't render correctly
locally as the image would be in a separate location.

A second issue also resolved by this commit is that, although you can
specify WALKTHROUGH_LOCATIONS it does not reflect in the web app.

The changes made in this commit provide a dynamic static route in
the server to allow for the files of each walkthrough to be served
using the same path as if you were previewing them locally. For
example: image::my.png will now work both locally and in the web app
(assuming that my.png is in the same directory as the adoc file).

The frontend has been updated to accomodate for these changes. Using
the new endpoint instead of the static /public endpoints from before.

Finally, the images from /public/images have been moved into their
respective walkthroughs and placed in an images/ subdirectory.

### CODE REVIEW

**Comment by aidenkeating** [comment]
@pb82 Would you mind taking a look? Some verification steps:

* Ensure that a default run `yarn start:dev` renders images correctly for each walkthrough.
* Clone `public/walkthroughs` out to a temporary dir.
* Restart the server, pointing at the temp dir e.g. `WALKTHROUGH_LOCATIONS=/tmp/walkthroughs yarn start:dev`. 
* Make a change to a walkthrough in the temp dir (add a new image for example) and ensure it's reflected in the web app after a refresh. This ensures the web app isn't just using the default walkthroughs.

**Comment by pb82** [comment]
👀

File: server.js [hunk 1]
```diff
@@ -250,7 +264,8 @@ function getWalkthroughInfoFromAdoc(id, dirName, doc) {
     shortDescription,
     time: getTotalWalkthroughTime(doc),
     adoc: path.join(dirName, 'walkthrough.adoc'),
-    json: path.join(dirName, 'walkthrough.json')
+    json: path.join(dirName, 'walkthrough.json'),
+    basePath: path.join(__dirname, dirName)
```
[the remark this answers is not in the document]
Comment (aidenkeating) [reply server.js]: Oh good catch!

File: server.js [hunk 2]
```diff
@@ -74,6 +74,22 @@ app.get('/customConfig', (req, res) => {
...
+    params: { walkthroughId }
+  } = req;
+  const file = req.param(0);
+  const walkthrough = walkthroughs.find(wt => wt.id === walkthroughId);
+  if (!walkthrough) {
+    return res.status(404).json({ error: `Walkthrough with ID ${walkthroughId} is not found` });
+  }
+  // Dotpaths are not allowed by default, meaning an end-user shouldn't be able
+  // to abuse the file system using the wildcard file param.
+  res.sendFile(`${walkthrough.basePath}/${file}`);
```
Comment (pb82) [inline server.js]: See below comment for why it's best to change this to something like here: https://github.com/integr8ly/tutorial-web-app/pull/315/files#diff-78c12f5adc1848d13b1c6f07055d996eR101

**Pull request merged**